openstack/baremetal/v1/nodes/requests.go

```diff
@@ -408,11 +408,19 @@ type ProvisionStateOptsBuilder interface {
 	ToProvisionStateMap() (map[string]interface{}, error)
 }
 
+// Starting with Ironic API version 1.56, a configdrive may be a JSON object with structured data.
+// Prior to this version, it must be a base64-encoded, gzipped ISO9660 image.
+type ConfigDrive struct {
+	MetaData    map[string]interface{} `json:"meta_data,omitempty"`
+	NetworkData map[string]interface{} `json:"network_data,omitempty"`
+	UserData    interface{}            `json:"user_data,omitempty"`
+}
+
 // ProvisionStateOpts for a request to change a node's provision state. A config drive should be base64-encoded
 // gzipped ISO9660 image.
 type ProvisionStateOpts struct {
 	Target         TargetProvisionState `json:"target" required:"true"`
-	ConfigDrive    string               `json:"configdrive,omitempty"`
+	ConfigDrive    interface{}          `json:"configdrive,omitempty"`
```

Comment:
I've changed this to `interface{}` so as not to break any code using just a `string`, since I can't implement an interface on a primitive type like `ConfigDriveOptsBuilder` - does that make sense?

This `configdrive` field is even now way more overloaded as you can see in the docs as it can be 3 different things, and with the JSON option,  the user_data field in the JSON can be a map, an array, or a string itself.

Reply:
If it helps, Gophercloud provides no backwards compatibility guarantees - there's way too much still being implemented (as well as learning how to implement it) to be considered stable.

I'm sure the amount of people using this piece of code is very small.

If it's easier to refactor this as a breaking change, go for it.

Reply:
I'm not sure it's easier - just trying to figure out what the most idiomatic thing to do is.  `interface{}` is nice here since it works with the myriad of new possibilities, but I realize it doesn't give the user a hint about what to send.

I could do an opts builder, but doing that with primitive types like string (`type ConfigDriveString string`), is awkward since you always have to cast. Is that how people tend to do it?

Reply:
How about pushing this off to the `gophercloud/utils` config drive stuff? I might be wrong, but it looks like it could handle this with just a little modification. For example, if `BuildDirectory` wasn't specified, then assume the user doesn't want to build an ISO and instead just generate the JSON structure and return the marshalled string.

Or add a new boolean field called `BuildISO` that would explicitly toggle either building an ISO or returning a JSON string.

Reply:
err - actually returning a marshalled string would be, well, a string and not a JSON data structure. One way around this would be to change the field from `string` to `interface{}`, just as you're doing, but assume the user is always using the utility function to generate a config drive value. I think that would work.

I agree with your comment about not giving a user a hint, but this feels like a special case to me. Keeping it open-ended with `interface{}` and provide a convenience utility to assist with creating a config drive seems like a good solution to this situation. Using `interface{}` as the value also allows the user to choose not to use the `gophercloud/utils` stuff and generate the configdrive however they want.

Reply:
Sorry for the delay in getting back to this.  Thanks for the comments.  So could this PR remain the way it is? 

I could move this struct to utils, but this is officially part of the Ironic API and the expectation is that the vast majority would use this mechanism for the config drive.  There's not a lot of reasons to do it any other way.  That's the strongest argument I think to keep it here.

Users of the older API could use the gophercloud/utils tooling, and advanced users could just build their own (as this would remain `interface{}`).  What do you think?

Reply:
Sure, that sounds good to me. Technically the code is correct - I'll defer to your expertise on what's best for the implementation.

In this case, this PR is good to go?

Reply:
Yes, it's ready - thanks!

Do you know why Travis failed? Looks like something with `go get`.

Reply:
There appears to be an issue with the latest master of Go. I'll have to look into this before merging.

Reply:
Fortunately it was fixed upstream quite quickly.